### PR TITLE
Resolves #258 : dockerized front

### LINKED ISSUE #258: [ Docker ] Dockerized front with github action
- [x] Dockerized front
- [x] Add a github action who push new image at every new merged commit

### PR DESCRIPTION
**Pull request by yannis-mlgrn**
Fixed #258

### CODE REVIEW

**Comment by yannis-mlgrn** [comment]
see https://hub.docker.com/repository/docker/yannismlgrn/donkeycarlph-front/general

**Review by Benvii — CHANGES_REQUESTED** [review]
Thanks for your PR. Some mistakes see comments.

## Wrong node version issue 
During the build stage I noticed this warning : 
```bash
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!

```

According to [this article](https://www.abrahamberg.com/blog/npm-package-json-lock-version-1-or-2/) you need for lock package v2 : 
```
lockfileVersion: 2 => npm v7+, which is backwards compatible to v1 lockfiles.
``` 
Which is included from node version `Node.js 15.14.0` (npm v7.7.6) according to this version table : https://nodejs.org/fr/download/releases

So you need to update node version (which is currently v14), @Nils-Abjorn will certainly agree we should you node latest LTS version which is Node.js 18.16.0	Hydrogen.

File: .github/workflows/front-push-docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Build and Push Docker Image to Docker Hub
+
+on: 
+  pull_request_target: # Allow from forks, but a condition limit where the forks comes from on the sensitive job
+    paths:
+    - donkeycarManagerFront/**
```
[the remark this answers is not in the document]
Comment (yannis-mlgrn) [reply .github/workflows/front-push-docker-image.yml]: done at [d585cb1](https://github.com/mdl29/donkeycarLPH/pull/259/commits/d585cb16bc488d2c7f490b359d8c0c6542f40266)

File: donkeycarManagerFront/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Donkeycar front build
+FROM node:18 AS build-stage
+COPY package*.json ./
+RUN npm install       
+COPY . .
+WORKDIR ./donkeycarManagerFront
+RUN  npm install @vue/cli-service --save-dev
+RUN npm run build
```
Comment (Benvii) [inline donkeycarManagerFront/Dockerfile:8]: It fail, seems to work only with the flag `--legacy-peer-deps`, this means that the lock file needs to be updated / change.

```
docker build -t benvii/donkeycarlph-front .
Sending build context to Docker daemon  173.7MB
Step 1/11 : FROM node:18 AS build-stage
18: Pulling from library/node
bd73737482dd: Pull complete 
6710592d62aa: Pull complete 
75256935197e: Pull complete 
c1e5026c6457: Pull complete 
f2e4b4cbd0b8: Pull complete 
06cfeca21bcd: Pull complete 
cb6259d61ade: Pull complete 
e7ad8f1516a1: Pull complete 
Digest: sha256:8cdf7234449f35e579f7491137807ddb3a089f028862f7ab69af437cc9f47ff1
Status: Downloaded newer image for node:18
 ---> 78b037dbb659
Step 2/11 : COPY package*.json ./
 ---> d1b541b5545c
Step 3/11 : RUN npm install
 ---> Running in 280ec8658419
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @vue/eslint-config-standard@6.1.0
npm ERR! Found: eslint-plugin-vue@8.7.1
npm ERR! node_modules/eslint-plugin-vue
npm ERR!   dev eslint-plugin-vue@"^8.0.3" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer eslint-plugin-vue@"^7.0.0" from @vue/eslint-config-standard@6.1.0
npm ERR! node_modules/@vue/eslint-config-standard
npm ERR!   dev @vue/eslint-config-standard@"^6.1.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: eslint-plugin-vue@7.20.0
npm ERR! node_modules/eslint-plugin-vue
npm ERR!   peer eslint-plugin-vue@"^7.0.0" from @vue/eslint-config-standard@6.1.0
npm ERR!   node_modules/@vue/eslint-config-standard
npm ERR!     dev @vue/eslint-config-standard@"^6.1.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /root/.npm/_logs/2023-06-10T16_12_07_615Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2023-06-10T16_12_07_615Z-debug-0.log
The command '/bin/sh -c npm install' returned a non-zero code: 1

```

**Review by Benvii — CHANGES_REQUESTED** [review]
Sorry but you didn't test it and it simply doesn't work

File: donkeycarManagerFront/Dockerfile (hunk 2 above)
Comment (yannis-mlgrn) [reply donkeycarManagerFront/Dockerfile:8]: Fixe here [ad96fa4](https://github.com/mdl29/donkeycarLPH/pull/259/commits/ad96fa48071154afea43dbdba608c982b84883b7)

**Review by Benvii — APPROVED** [review]

**Review by Benvii — APPROVED** [review]

**Pull request merged**